CRAN WARN on DEVEL

From Kurt:

> Dear maintainer,
> 
> Please see the problems shown on
> <https://cran.r-project.org/web/checks/check_results_rrum.html>.
> 
> Please correct before 2023-12-12 to safely retain your package on CRAN.

CRAN check:

```
Version: 0.2.0
Check: whether package can be installed
Result: WARN 
  Found the following significant warnings:
    RcppExports.cpp:48:18: warning: format string is not a string literal (potentially insecure) [-Wformat-security]
  See ‘/home/hornik/tmp/R.check/r-devel-clang/Work/PKGS/rrum.Rcheck/00install.out’ for details.
  * used C++ compiler: ‘Debian clang version 17.0.5 (1)’
Flavor: [r-devel-linux-x86_64-debian-clang](https://www.r-project.org/nosvn/R.check/r-devel-linux-x86_64-debian-clang/rrum-00check.html)

Version: 0.2.0
Check: C++ specification
Result: NOTE 
    Specified C++11: please drop specification unless essential
Flavors: [r-devel-linux-x86_64-debian-clang](https://www.r-project.org/nosvn/R.check/r-devel-linux-x86_64-debian-clang/rrum-00check.html), [r-devel-linux-x86_64-debian-gcc](https://www.r-project.org/nosvn/R.check/r-devel-linux-x86_64-debian-gcc/rrum-00check.html), [r-devel-linux-x86_64-fedora-clang](https://www.r-project.org/nosvn/R.check/r-devel-linux-x86_64-fedora-clang/rrum-00check.html), [r-devel-linux-x86_64-fedora-gcc](https://www.r-project.org/nosvn/R.check/r-devel-linux-x86_64-fedora-gcc/rrum-00check.html), [r-devel-windows-x86_64](https://www.r-project.org/nosvn/R.check/r-devel-windows-x86_64/rrum-00check.html), [r-patched-linux-x86_64](https://www.r-project.org/nosvn/R.check/r-patched-linux-x86_64/rrum-00check.html), [r-release-linux-x86_64](https://www.r-project.org/nosvn/R.check/r-release-linux-x86_64/rrum-00check.html), [r-release-macos-arm64](https://www.r-project.org/nosvn/R.check/r-release-macos-arm64/rrum-00check.html), [r-release-macos-x86_64](https://www.r-project.org/nosvn/R.check/r-release-macos-x86_64/rrum-00check.html), [r-release-windows-x86_64](https://www.r-project.org/nosvn/R.check/r-release-windows-x86_64/rrum-00check.html)

Version: 0.2.0
Check: Rd \usage sections
Result: NOTE 
  Documented arguments not in \usage in Rd file 'rrum_helper.Rd':
    ‘deltas’
  
  Functions with \usage entries need to have the appropriate \alias
  entries, and all their arguments documented.
  The \usage entries must correspond to syntactically valid R code.
  See chapter ‘Writing R documentation files’ in the ‘Writing R
  Extensions’ manual.
Flavors: [r-devel-linux-x86_64-debian-clang](https://www.r-project.org/nosvn/R.check/r-devel-linux-x86_64-debian-clang/rrum-00check.html), [r-devel-linux-x86_64-debian-gcc](https://www.r-project.org/nosvn/R.check/r-devel-linux-x86_64-debian-gcc/rrum-00check.html), [r-devel-linux-x86_64-fedora-clang](https://www.r-project.org/nosvn/R.check/r-devel-linux-x86_64-fedora-clang/rrum-00check.html), [r-devel-linux-x86_64-fedora-gcc](https://www.r-project.org/nosvn/R.check/r-devel-linux-x86_64-fedora-gcc/rrum-00check.html), [r-devel-windows-x86_64](https://www.r-project.org/nosvn/R.check/r-devel-windows-x86_64/rrum-00check.html)
```

--- a/.Rbuildignore
+++ b/.Rbuildignore
@@ -12,3 +12,4 @@ man-roxygen
 ^docs$
 ^pkgdown$
 ^\.github$
+^CRAN-SUBMISSION$

--- a/.gitignore
+++ b/.gitignore
@@ -4,3 +4,4 @@ docs/
 CRAN-RELEASE
 docs
 .DS_Store
+CRAN-SUBMISSION

--- a/DESCRIPTION
+++ b/DESCRIPTION
@@ -26,7 +26,7 @@ URL: https://tmsalab.github.io/rrum/, https://github.com/tmsalab/rrum
 BugReports: https://github.com/tmsalab/rrum/issues
 License: GPL (>= 2)
 Depends: R (>= 4.1.0), simcdm (>= 0.1.0)
-Imports: Rcpp  (>= 1.0.11)
+Imports: Rcpp (>= 1.0.11)
 LinkingTo: Rcpp, RcppArmadillo (>= 0.12.6.6.0), rgen, simcdm
 Suggests: 
     testthat,

--- a/DESCRIPTION
+++ b/DESCRIPTION
@@ -22,16 +22,15 @@ Authors@R: c(
 Description: Implementation of Gibbs sampling algorithm for Bayesian Estimation
     of the Reduced Reparameterized Unified Model ('rrum'), described by 
     Culpepper and Hudson (2017) <doi: 10.1177/0146621617707511>.
-URL: https://tmsalab.github.io/rrum, https://github.com/tmsalab/rrum
+URL: https://tmsalab.github.io/rrum/, https://github.com/tmsalab/rrum
 BugReports: https://github.com/tmsalab/rrum/issues
 License: GPL (>= 2)
-Depends: R (>= 3.4.0), simcdm (>= 0.1.0)
-Imports: Rcpp (>= 1.0.0)
-LinkingTo: Rcpp, RcppArmadillo (>= 0.9.200), rgen, simcdm
+Depends: R (>= 4.1.0), simcdm (>= 0.1.0)
+Imports: Rcpp  (>= 1.0.11)
+LinkingTo: Rcpp, RcppArmadillo (>= 0.12.6.6.0), rgen, simcdm
 Suggests: 
     testthat,
     covr
-SystemRequirements: C++11
 Encoding: UTF-8
-RoxygenNote: 7.1.1
+RoxygenNote: 7.2.3
 Roxygen: list(markdown = TRUE)

--- a/NEWS.md
+++ b/NEWS.md
@@ -1,8 +1,20 @@
+# rrum 0.2.2
+
+## Bugfixes
+
+- Addressed two issues in R-devel related to C++ usage (deprecation of `CXX_STD` and attribute compilation woes). ([#15](https://github.com/tmsalab/rrum/issues/15))
+- Changed the format in `CITATION` to use `c()` instead of `as.personList()` and
+  switched entries from `citEntry()` to `bibentry()`.
+
+## Deployment
+
+- Updated GitHub Actions for testing, pkgdown, and coverage ([#16](https://github.com/tmsalab/rrum/pull/16)).
+
 # rrum 0.2.1
 
 ## Documentation
 
-- Added a `pkgdown` website that deploys to <http://tmsalab.github.io/rrum> ([#12](https://github.com/tmsalab/rrum/pull/12), [#14](https://github.com/tmsalab/rrum/pull/14)).
+- Added a `pkgdown` website that deploys to <https://tmsalab.github.io/rrum/> ([#12](https://github.com/tmsalab/rrum/pull/12), [#14](https://github.com/tmsalab/rrum/pull/14)).
 - Pruned non-user facing documentation entry ([#14](https://github.com/tmsalab/rrum/pull/14)).
 
 ## Deployment

--- a/R/RcppExports.R
+++ b/R/RcppExports.R
@@ -14,6 +14,7 @@
 #'                     of the items.An entry of 1 indicates attribute \eqn{k}
 #'                     is required to answer item \eqn{j}. An entry of one
 #'                     indicates attribute \eqn{k} is not required.
+#' @param delta0       A `vector`, parameters for the Dirichlet prior on `pi`.
 #' @param chain_length A `numeric` indicating the number of iterations of
 #'                     Gibbs sampler to be run.  Default is set to 10000.
 #' @param as           A `numeric`, parameter for the prior distribution of
@@ -28,7 +29,6 @@
 #' @param bg           A `numeric`, parameter for the prior distribution of
 #'                     `pistar`.  High values as encourage lower values of
 #'                     `rstar`.
-#' @param deltas       A `vector`, parameters for the Dirichlet prior on `pi`.
 #'
 #' @return A `List`
 #'
@@ -57,5 +57,5 @@ rrum_helper <- function(Y, Q, delta0, chain_length = 10000L, as = 1, bs = 1, ag 
 
 # Register entry points for exported C++ functions
 methods::setLoadAction(function(ns) {
-    .Call('_rrum_RcppExport_registerCCallable', PACKAGE = 'rrum')
+    .Call(`_rrum_RcppExport_registerCCallable`)
 })

--- a/README.Rmd
+++ b/README.Rmd
@@ -15,13 +15,12 @@ knitr::opts_chunk$set(
 # rrum
 
 <!-- badges: start -->
-[![R build status](https://github.com/tmsalab/rrum/workflows/R-CMD-check/badge.svg)](https://github.com/tmsalab/rrum/actions)
-[![Package-License](http://img.shields.io/badge/license-GPL%20(%3E=2)-brightgreen.svg?style=flat)](http://www.gnu.org/licenses/gpl-2.0.html)
-[![CRAN Version Badge](http://www.r-pkg.org/badges/version/rrum)](https://cran.r-project.org/package=rrum)
-[![CRAN Status](https://cranchecks.info/badges/worst/rrum)](https://cran.r-project.org/web/checks/check_results_rrum.html)
-[![RStudio CRAN Mirror's Monthly Downloads](http://cranlogs.r-pkg.org/badges/rrum?color=brightgreen)](http://www.r-pkg.org/pkg/rrum)
-[![RStudio CRAN Mirror's Total Downloads](http://cranlogs.r-pkg.org/badges/grand-total/rrum?color=brightgreen)](http://www.r-pkg.org/pkg/rrum)
-[![Coverage test coverage](https://codecov.io/gh/tmsalab/rrum/branch/master/graph/badge.svg)](https://codecov.io/github/tmsalab/rrum?branch=master)
+[![R-CMD-check](https://github.com/tmsalab/rrum/actions/workflows/R-CMD-check.yaml/badge.svg)](https://github.com/tmsalab/rrum/actions/workflows/R-CMD-check.yaml)
+[![Package-License](https://img.shields.io/badge/license-GPL%20(%3E=2)-brightgreen.svg?style=flat)](https://www.gnu.org/licenses/gpl-2.0.html)
+[![CRAN Version Badge](https://www.r-pkg.org/badges/version/rrum)](https://cran.r-project.org/package=rrum)
+[![CRAN Status](https://badges.cranchecks.info/worst/rrum.svg)](https://cran.r-project.org/web/checks/check_results_rrum.html)
+[![RStudio CRAN Mirror's Monthly Downloads](https://cranlogs.r-pkg.org/badges/rrum?color=brightgreen)](https://www.r-pkg.org/pkg/rrum)
+[![RStudio CRAN Mirror's Total Downloads](https://cranlogs.r-pkg.org/badges/grand-total/rrum?color=brightgreen)](https://www.r-pkg.org/pkg/rrum)
 <!-- badges: end -->
 
 The goal of `rrum` is to provide an implementation of Gibbs sampling algorithm 
@@ -32,34 +31,44 @@ described by Culpepper and Hudson (2017) <doi: 10.1177/0146621617707511>.
 
 You can install `rrum` from CRAN using:
 
-```{r cran-installation, eval = FALSE}
+```{r}
+#| label: cran-installation
+#| eval: false
 install.packages("rrum")
 ```
 
 Or, you can be on the cutting-edge development version on GitHub using:
 
-```{r gh-installation, eval = FALSE}
-if(!requireNamespace("devtools")) install.packages("devtools")
-devtools::install_github("tmsalab/rrum")
+```{r}
+#| label: gh-installation
+#| eval: false
+# install.packages('remotes')
+remotes::install_github("tmsalab/rrum")
 ```
 
 ## Usage
 
 To use `rrum`, load the package using:
 
-```{r example}
+```{r}
+#| label: example
+#| eval: false
 library("rrum")
 ```
 
 From here, the rRUM model can be estimated using:
 
-```{r rrum-est, eval = FALSE}
+```{r}
+#| label: rrum-est
+#| eval: false
 rrum_model = rrum(<data>, <q>)
 ```
 
 Additional parameters can be accessed with:
 
-```{r rrum-est-adv, eval = FALSE}
+```{r}
+#| label: rrum-est-adv
+#| eval: false
 rrum_model = rrum(<data>, <q>, chain_length = 10000L,
                   as = 1, bs = 1, ag = 1, bg = 1,
                   delta0 = rep(1, 2^ncol(Q)))
@@ -67,7 +76,9 @@ rrum_model = rrum(<data>, <q>, chain_length = 10000L,
 
 `rRUM` item data can be simulated using:
 
-```{r rrum-sim, eval = FALSE}
+```{r}
+#| label: rrum-sim
+#| eval: false
 # Set a seed for reproducibility
 set.seed(888)
 
@@ -106,7 +117,9 @@ To ensure future development of the package, please cite `rrum`
 package if used during an analysis or simulation study. Citation information
 for the package may be acquired by using in *R*:
 
-```{r, eval = FALSE}
+```{r}
+#| label: citation
+#| eval: false
 citation("rrum")
 ```
 

--- a/README.md
+++ b/README.md
@@ -5,19 +5,16 @@
 
 <!-- badges: start -->
 
-[![R build
-status](https://github.com/tmsalab/rrum/workflows/R-CMD-check/badge.svg)](https://github.com/tmsalab/rrum/actions)
-[![Package-License](http://img.shields.io/badge/license-GPL%20\(%3E=2\)-brightgreen.svg?style=flat)](http://www.gnu.org/licenses/gpl-2.0.html)
+[![R-CMD-check](https://github.com/tmsalab/rrum/actions/workflows/R-CMD-check.yaml/badge.svg)](https://github.com/tmsalab/rrum/actions/workflows/R-CMD-check.yaml)
+[![Package-License](https://img.shields.io/badge/license-GPL%20(%3E=2)-brightgreen.svg?style=flat)](https://www.gnu.org/licenses/gpl-2.0.html)
 [![CRAN Version
-Badge](http://www.r-pkg.org/badges/version/rrum)](https://cran.r-project.org/package=rrum)
+Badge](https://www.r-pkg.org/badges/version/rrum)](https://cran.r-project.org/package=rrum)
 [![CRAN
-Status](https://cranchecks.info/badges/worst/rrum)](https://cran.r-project.org/web/checks/check_results_rrum.html)
+Status](https://badges.cranchecks.info/worst/rrum.svg)](https://cran.r-project.org/web/checks/check_results_rrum.html)
 [![RStudio CRAN Mirror’s Monthly
-Downloads](http://cranlogs.r-pkg.org/badges/rrum?color=brightgreen)](http://www.r-pkg.org/pkg/rrum)
+Downloads](https://cranlogs.r-pkg.org/badges/rrum?color=brightgreen)](https://www.r-pkg.org/pkg/rrum)
 [![RStudio CRAN Mirror’s Total
-Downloads](http://cranlogs.r-pkg.org/badges/grand-total/rrum?color=brightgreen)](http://www.r-pkg.org/pkg/rrum)
-[![Coverage test
-coverage](https://codecov.io/gh/tmsalab/rrum/branch/master/graph/badge.svg)](https://codecov.io/github/tmsalab/rrum?branch=master)
+Downloads](https://cranlogs.r-pkg.org/badges/grand-total/rrum?color=brightgreen)](https://www.r-pkg.org/pkg/rrum)
 <!-- badges: end -->
 
 The goal of `rrum` is to provide an implementation of Gibbs sampling
@@ -36,8 +33,8 @@ install.packages("rrum")
 Or, you can be on the cutting-edge development version on GitHub using:
 
 ``` r
-if(!requireNamespace("devtools")) install.packages("devtools")
-devtools::install_github("tmsalab/rrum")
+# install.packages('remotes')
+remotes::install_github("tmsalab/rrum")
 ```
 
 ## Usage
@@ -46,7 +43,6 @@ To use `rrum`, load the package using:
 
 ``` r
 library("rrum")
-#> Loading required package: simcdm
 ```
 
 From here, the rRUM model can be estimated using:

--- a/inst/CITATION
+++ b/inst/CITATION
@@ -1,8 +1,8 @@
 citHeader("To cite the 'rrum' R package and method paper in publications use:")
 
-citEntry(entry = "Manual",
+bibentry(bibtype = "Manual",
          title        = "{rrum: Bayesian Estimation of the Reduced Reparameterized Unified Model with Gibbs Sampling}",
-         author       = personList(as.person("Steven Andrew Culpepper"),
+         author       = c(as.person("Steven Andrew Culpepper"),
                                    as.person("Aaron Hudson"),
                                    as.person("James Joseph Balamuta")
                                   ),
@@ -12,7 +12,7 @@ citEntry(entry = "Manual",
                               "URL https://cran.r-project.org/package=rrum.")
 )
 
-citEntry(entry   = "Article",
+bibentry(bibtype   = "Article",
          title   = "{An Improved Strategy for Bayesian Estimation of the Reduced Reparameterized Unified Model}",
          author  = personList(as.person("Steven Andrew Culpepper"),
                               as.person("Aaron Hudson")),

--- a/man/rrum-package.Rd
+++ b/man/rrum-package.Rd
@@ -6,9 +6,7 @@
 \alias{_PACKAGE}
 \title{rrum: Bayesian Estimation of the Reduced Reparameterized Unified Model with Gibbs Sampling}
 \description{
-Implementation of Gibbs sampling algorithm for Bayesian Estimation
-    of the Reduced Reparameterized Unified Model ('rrum'), described by 
-    Culpepper and Hudson (2017) <doi: 10.1177/0146621617707511>.
+Implementation of Gibbs sampling algorithm for Bayesian Estimation of the Reduced Reparameterized Unified Model ('rrum'), described by Culpepper and Hudson (2017) \doi{ 10.1177/0146621617707511}.
 }
 \details{
 Implemention of a Bayesian estimation for the
@@ -27,7 +25,7 @@ Carolina.
 \seealso{
 Useful links:
 \itemize{
-  \item \url{https://tmsalab.github.io/rrum}
+  \item \url{https://tmsalab.github.io/rrum/}
   \item \url{https://github.com/tmsalab/rrum}
   \item Report bugs at \url{https://github.com/tmsalab/rrum/issues}
 }

--- a/src/Makevars
+++ b/src/Makevars
@@ -1,6 +1,3 @@
-# Enable C++11
-CXX_STD=CXX11
-
 # Enable the header file and OpenMP
 PKG_CXXFLAGS=-I../inst/include $(SHLIB_OPENMP_CXXFLAGS) 
 

--- a/src/Makevars.win
+++ b/src/Makevars.win
@@ -1,6 +1,3 @@
-# Enable C++11
-CXX_STD=CXX11
-
 # Enable the header file and OpenMP
 PKG_CXXFLAGS=-I../inst/include $(SHLIB_OPENMP_CXXFLAGS) 
 

--- a/src/RcppExports.cpp
+++ b/src/RcppExports.cpp
@@ -9,6 +9,11 @@
 
 using namespace Rcpp;
 
+#ifdef RCPP_USE_GLOBAL_ROSTREAM
+Rcpp::Rostream<true>&  Rcpp::Rcout = Rcpp::Rcpp_cout_get();
+Rcpp::Rostream<false>& Rcpp::Rcerr = Rcpp::Rcpp_cerr_get();
+#endif
+
 // rrum_helper
 Rcpp::List rrum_helper(const arma::mat& Y, const arma::mat& Q, const arma::vec& delta0, unsigned int chain_length, double as, double bs, double ag, double bg);
 static SEXP _rrum_rrum_helper_try(SEXP YSEXP, SEXP QSEXP, SEXP delta0SEXP, SEXP chain_lengthSEXP, SEXP asSEXP, SEXP bsSEXP, SEXP agSEXP, SEXP bgSEXP) {
@@ -45,7 +50,7 @@ RcppExport SEXP _rrum_rrum_helper(SEXP YSEXP, SEXP QSEXP, SEXP delta0SEXP, SEXP 
     if (rcpp_isError_gen) {
         SEXP rcpp_msgSEXP_gen = Rf_asChar(rcpp_result_gen);
         UNPROTECT(1);
-        Rf_error(CHAR(rcpp_msgSEXP_gen));
+        Rf_error("%s", CHAR(rcpp_msgSEXP_gen));
     }
     UNPROTECT(1);
     return rcpp_result_gen;

--- a/src/rRUM_Gibbs.cpp
+++ b/src/rRUM_Gibbs.cpp
@@ -185,6 +185,7 @@ Rcpp::List rrum_main(const arma::mat &Y, const arma::mat &Q,
 //'                     of the items.An entry of 1 indicates attribute \eqn{k}
 //'                     is required to answer item \eqn{j}. An entry of one
 //'                     indicates attribute \eqn{k} is not required.
+//' @param delta0       A `vector`, parameters for the Dirichlet prior on `pi`.
 //' @param chain_length A `numeric` indicating the number of iterations of
 //'                     Gibbs sampler to be run.  Default is set to 10000.
 //' @param as           A `numeric`, parameter for the prior distribution of
@@ -199,7 +200,6 @@ Rcpp::List rrum_main(const arma::mat &Y, const arma::mat &Q,
 //' @param bg           A `numeric`, parameter for the prior distribution of
 //'                     `pistar`.  High values as encourage lower values of
 //'                     `rstar`.
-//' @param deltas       A `vector`, parameters for the Dirichlet prior on `pi`.
 //'
 //' @return A `List`
 //'

